meson.build

```diff
@@ -0,0 +1,797 @@
+libcapstone_is_enabled = libcapstone_dep.found()
+if libcapstone_is_enabled
+    common_cflags += '-DHAVE_LIBCAPSTONE'
+endif
+
+##############################
+# -Dwith-libdw  build option #
+# -Dwith-libelf build option #
+##############################
+
+libelf_dep = dependency('libelf', required: get_option('with-libelf'))
+libelf_is_enabled = libelf_dep.found()
+if libelf_is_enabled
+    common_cflags += '-DHAVE_LIBELF'
+
+    libdw_dep = dependency('libdw', required: get_option('with-libdw'))
+    libdw_is_enabled = libdw_dep.found()
+    if libdw_is_enabled
+        common_cflags += '-DHAVE_LIBDW'
+    endif
+else
+    libdw_dep = []
+    libdw_is_enabled = false
+endif
+
+#################################
+# -Dwith-libluajit build option #
+#################################
+
+libluajit_dep = dependency('luajit', required: get_option('with-libluajit'))
+libluajit_is_enabled = libluajit_dep.found()
+if libluajit_is_enabled
+    common_cflags += '-DHAVE_LIBLUAJIT'
+endif
+
+###################################
+# -Dwith-libncursesw build option #
+###################################
+
+libncursesw_dep = dependency('ncursesw', required: get_option('with-libncursesw'))
+libncursesw_is_enabled = libncursesw_dep.found()
+if libncursesw_is_enabled
+    common_cflags += '-DHAVE_LIBNCURSES'
+endif
+
+#################################
+# -Dwith-libpython build option #
+#################################
+
+python_installation = python_module.find_installation('python', required: get_option('with-libpython'))
+libpython_dep = []
+libpython_is_enabled = python_installation.found()
+if libpython_is_enabled
+    libpython_dep = python_installation.dependency()
+    python_version = python_installation.language_version()
+    common_cflags += '-DLIBPYTHON_VERSION=@0@'.format(python_version)
+    common_cflags += '-DHAVE_LIBPYTHON' + python_version[0]
+endif
+
+##################################
+# -Dwith-libunwind build option #
+##################################
+
+libunwind_dep = dependency('libunwind', required: get_option('with-libunwind'))
+libunwind_is_enabled = get_option('debug') and libunwind_dep.found()
+if libunwind_is_enabled
+    common_cflags += '-DHAVE_LIBUNWIND'
+endif
+
+#################################
+# -Dwith-libstdcxx build option #
+#################################
+
+cxa_demangle_is_enabled = false
+cxa_demangle_dep = []
+have_cxa_demangle_code = run_command('cat', files('check-deps/__cxa_demangle.c')).stdout().strip()
+have_cxa_demangle_compiles = cc.compiles(have_cxa_demangle_code, name: 'cxa_demangle check', args: ['-lstdc++'])
+if get_option('with-libstdcxx').auto()
+    if have_cxa_demangle_compiles
+        cxa_demangle_dep = cc.find_library('stdc++', required: true)
+        cxa_demangle_is_enabled = true
+        common_cflags += '-DHAVE_CXA_DEMANGLE'
+        message('Enabling libstdc++')
+    else
+        message('Disabling libstdc++')
+    endif
+elif get_option('with-libstdcxx').enabled()
```

Comment:
This is duplicating additional logic around. What about:

```meson
cxa_demangle_dep = cc.find_library('stdc++', required: get_option('with-libstdcxx'))

if get_option('with-libstdcxx').enabled() and not have_cxa_demangle_compiles
    error('Failed to compile and run check-deps/__cxa_demangle.c')
elif cxa_demangle_dep.found() and have_cxa_demangle_compiles
    cxa_demangle_is_enabled = true
    common_cflags += '-DHAVE_CXA_DEMANGLE'
    msg = 'Enabling libstdc++'
    if get_option('with-libstdcxx').enabled()
        msg += ' (enabled by: with-libstdcxx)'
    endif
else
    msg = 'Disabling libstdc++'
    if get_option('with-libstdcxx').disabled()
        msg += ' (disabled by: with-libstdcxx)'
    elif cxa_demangle_dep.found()
        msg = ' (disabled by: failure to compile and run check-deps/__cxa_demangle.c)'
    endif
endif
message(msg) 
```

3 fewer lines, I added additional logging in the disabling case, the coding flow is less duplicative, and honestly I'd skip the last 13 lines:

```meson
cxa_demangle_dep = cc.find_library('stdc++', required: get_option('with-libstdcxx'))

if get_option('with-libstdcxx').enabled() and not have_cxa_demangle_compiles
    error('Failed to compile and run check-deps/__cxa_demangle.c')
elif cxa_demangle_dep.found() and have_cxa_demangle_compiles
    cxa_demangle_is_enabled = true
    common_cflags += '-DHAVE_CXA_DEMANGLE'
endif
```